api/apps/api/src/modules/projects/blm/change-blm-range.handler.ts

```diff
@@ -1,18 +1,82 @@
 import { CommandHandler, IInferredCommandHandler } from '@nestjs/cqrs';
-import { Either, right } from 'fp-ts/Either';
+import { Either, isLeft, left, right } from 'fp-ts/Either';
 
-import { ChangeBlmRange, ChangeRangeErrors } from './change-blm-range.command';
+import {
+  ChangeBlmRange,
+  ChangeRangeErrors,
+  queryFailure,
+} from './change-blm-range.command';
 import { ProjectBlmRepo } from '@marxan-api/modules/blm';
+import { InjectEntityManager, InjectRepository } from '@nestjs/typeorm';
+import { Project } from '@marxan-api/modules/projects/project.api.entity';
+import { EntityManager, Repository } from 'typeorm';
+import { DbConnections } from '@marxan-api/ormconfig.connections';
+import { Logger } from '@nestjs/common';
+import { SetProjectBlm } from '@marxan-api/modules/projects/blm/set-project-blm';
+import { BlmValuesCalculator } from '@marxan-api/modules/projects/blm/domain/blm-values-calculator';
 
 @CommandHandler(ChangeBlmRange)
 export class ChangeBlmRangeHandler
   implements IInferredCommandHandler<ChangeBlmRange> {
-  constructor(private readonly blmRepository: ProjectBlmRepo) {}
+  private readonly logger: Logger = new Logger(SetProjectBlm.name);
+
+  constructor(
+    @InjectRepository(Project)
+    protected readonly projectRepository: Repository<Project>,
+    @InjectEntityManager(DbConnections.geoprocessingDB)
+    private readonly entityManager: EntityManager,
+    private readonly blmRepository: ProjectBlmRepo,
+  ) {}
 
   async execute({
     projectId,
     range,
   }: ChangeBlmRange): Promise<Either<ChangeRangeErrors, true>> {
+    if (this.isInValidRange(range)) {
+      this.logger.error(
+        `Received range [${range[0]},${range[1]}] for project with ID: ${projectId} is invalid`,
+      );
+
+      return left(queryFailure);
+    }
+
+    const project = await this.projectRepository.findOneOrFail(projectId);
+    const area = await this.getPlanningUnitArea(project);
+
+    if (!area) {
+      this.logger.error(
+        `Could not get Planning Unit area for project with ID: ${projectId}`,
+      );
+
+      return left(queryFailure);
+    }
+
+    const blmValues = BlmValuesCalculator.with(range, area);
```

Comment:
I would recommend keeping NestJS DI system instead "hardcoded" utilities - otherwise we cannot change this behavior (or mock it)

Reply:
Initially this was hardcoded as a private method, but to reuse the logic I extracted it to domain service
(Further explained in the comment below)

Reply:
Given the comment below, if we keep it as "dependency", it could be enough:
` constructor( private readonly blmValuesGetter: Type = new BlmValuesCalculator )`

Reply:
@kgajowy  When passing it via constructor, NestJS tries to inject a provider and cannot resolve the dependencies. Should we move it to a provider even being a domain policy?

Reply:
@AaronPerezPerez we can make a pretty dummy factory which will construct such policy, what do you think?

Reply:
Great idea!